### PR TITLE
Do not run binaryen opts in -O1

### PR DESCRIPTION
**Pull request by kripken**
As suggested by @dschuff in https://github.com/emscripten-core/emscripten/pull/12173#pullrequestreview-487133883

This appears to regress code size by around 5-10% on average for `-O1`. Given
that this mode is kind of a "compromise" build anyhow, with most opts in return
for fast iteration, it seems like a better tradeoff to lose that 5-10% in return for
much faster link times - almost 2x faster - and also avoiding DWARF rewriting.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -533,23 +533,30 @@ def backend_binaryen_passes():
   # wasm backend output can benefit from the binaryen optimizer (in asm2wasm,
   # we run the optimizer during asm2wasm itself). use it, if not overridden.
 
+  # run the binaryen optimize in -O2+. in -O0 we don't need it obviously, while
```
Comment (dschuff) [inline emcc.py:536]:
```suggestion
  # run the binaryen optimizer in -O2+. in -O0 we don't need it obviously, while
```

File: emcc.py [hunk 2]
```diff
@@ -533,23 +533,30 @@ def backend_binaryen_passes():
...
   # safe heap must run before post-emscripten, so post-emscripten can apply the sbrk ptr
   if shared.Settings.SAFE_HEAP:
     passes += ['--safe-heap']
-  if shared.Settings.OPT_LEVEL > 0:
+  if run_binaryen_optimizer:
     passes += ['--post-emscripten']
     if not shared.Settings.EXIT_RUNTIME:
       passes += ['--no-exit-runtime']
-  if shared.Settings.OPT_LEVEL > 0 or shared.Settings.SHRINK_LEVEL > 0:
+  if run_binaryen_optimizer:
```
Comment (dschuff) [inline emcc.py:551]: you removed the check for SHRINK_LEVEL here but didn't add it back anywhere else. Should it go into `run_binaryen_optimizer`?

**Review by dschuff — APPROVED** [review]
10% size hit for a big compile performance win (and making it better for debugging) seems like a good tradeoff for O1 to me.

**Review by sbc100 — APPROVED** [review]

File: emcc.py (hunk 2 above)
Comment (kripken) [reply emcc.py:551]: This was intentional. `SHRINK_LEVEL` is for `-Os/-Oz` and in those modes we have `OPT_LEVEL >= 2` anyhow.

File: tools/building.py [hunk 3]
```diff
@@ -1579,7 +1579,7 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
       if shared.Settings.SUPPORT_LONGJMP:
         extra += '\nnote: to disable longjmp support (which requires changes after link) use -s SUPPORT_LONGJMP=0'
       if shared.Settings.OPT_LEVEL > 0:
-        extra += '\nnote: optimizations always require changes, build with -O0 instead'
+        extra += '\nnote: -O2+ optimizations always require changes, build with -O0 or -O1 instead'
```
Comment (dschuff) [inline tools/building.py:1582]: drive-by for lime 1578: it would be more clear to a non-emscripten developer if it said 'int64' legalization or something that mentioned that specifically.

**Review by dschuff — APPROVED** [review]

File: tools/building.py (hunk 3 above)
Comment (kripken) [reply tools/building.py:1582]: Good idea, added.

**Comment by bmeurer** [comment]
I love it, thanks!
